File: https-server.js
import { createServer } from "https"
import { readFileSync } from "fs"

// Read the SSL certificate files
const privateKey = readFileSync("./key.pem", "utf8")
const certificate = readFileSync("./cert.pem", "utf8")

export default (port = 3000) => {
  const server = createServer(
    { key: privateKey, cert: certificate },
    (req, res) => {
      res.writeHead(200, null, { "Content-type": "text/plain" })
      res.end("1")
    }
  ).listen(port)

  console.info("Started HTTPS server on port", port)

  return () => server.close()
}

File: servers/https-server.py
import http.server
import ssl
import sys
import socket

defaultPort = 3000

# Get the port from the command line arguments
port = int(sys.argv[1]) if len(sys.argv) > 1 else defaultPort

class MyHTTPSServer(http.server.HTTPServer):
    def server_bind(self):
        self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
        super().server_bind()

class MySimpleHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header("Content-type", "text/plain")
        self.end_headers()
        self.wfile.write(b"1")

    # Override this method to disable the default logging
    def log_message(self, format, *args):
        return

# Set up the SSL context using the key and cert files
context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
context.load_cert_chain(certfile='./cert.pem', keyfile='./key.pem')

httpd = MyHTTPSServer(('', port), MySimpleHTTPRequestHandler)
httpd.socket = context.wrap_socket(httpd.socket, server_side=True)
print(f"Started HTTPS server on port {port}")
httpd.serve_forever()
